lib/streamlit/elements/widgets/number_input.py

```diff
@@ -73,47 +74,229 @@ def deserialize(
 
 
 class NumberInputMixin:
+    # Ok hold your breath the typing overloads here are annnnnnnoying. Per the "number_input" docstring,
+    # the return value type depends on the provided types for the "value", "min_value", "max_value", and "step"
+    # arguments, all of which are optional and may be provided either as positional OR keyword arguments.
+    # In order to fully cover every possible call signature, we require ~14 different overload signatures.
+    # For easier readability, all the arguments with un-changing types across these signatures have been
+    # collapsed onto a single line.
+
+    # fmt: off
+    # min_value: int given as a positional OR keyword argument, value: "min" or not given
```

Comment:
Hm, do we need another case here for `min_value: int` being given as a keyword (/ not positional) arg? Not entirely sure if the set of overloads already covers that case somehow, but it is weird to me that it's missing given there's an overload for `max_value: int` as a kwarg

Reply:
Since `min_value` is the first argument with a default value, I don't think you need one. 

```python
number_input(min_value=5, label="foo")
```
should match both 
```python
    def number_input(
        self,
        label: str,
        min_value: int,
        ... (all other arguments with default values)
```
and
```python
    def number_input(
        self,
        label: str,
        *,
        min_value: int,
        ... (all other arguments with default values)
```
and 
```python
number_input(min_value=5)
```
is not allowed.

Reply:
Ah, ok. Yeah makes sense 👍